README.md

```diff
@@ -122,11 +122,39 @@ LOG_LEVEL=trace npm run test
 
 Alternatively, one may set `LOG_LEVEL` in the `.env` file.
 
-### Authentication
-
-The service validates a Bearer JSON Web Token (JWT) found in the `Authorization` HTTP header.
-
-See the `.env.example` for relevant variables.
+### Authentication and Authorization
+
+A valid Bearer JSON Web Token (Bearer JWT) is required in requests to the PDC service. The PDC officially reccomends using [KeyCloak](https://www.keycloak.org/) as the authentication provider. Please refer to [the relevant documentation for setup and configuration](https://www.keycloak.org/guides#getting-started)
+
+The relevant authorization enviroment variables as they appear in the `.env.example` file are:
+
+- `AUTH_SERVER_ISSUER`: the URL for the issuer (this is the "realm" URL in Keycloak terminology).
+- `OPENAPI_DOCS_AUTH_CLIENT_ID`: the public identifier for the app ("Client ID" field in Keycloack).
+
+#### An example with a Keycloak authentication and authorization service
+
+From the Keycloak admin interface, e.g. https://my-host/admin:
+
+1. Add a realm e.g. "pdc" (avoid spaces in the name).
+2. Within the realm just added in step (1) add a client e.g. "pdc-service", with Client authentication off and Authorization off:
+   - Standard flow unchecked,
+   - Direct access grants unchecked,
+   - Implicit flow unchecked,
+   - Service accounts roles unchecked,
+   - OAuth 2.0 Device Authorization Grant (optional), and
+   - OIDC CIBA Grant unchecked.
+3. In the settings for the client added in step (2), set the root URL and Home URL to the URL of the PDC service (not the auth service). Add redirect URIs:
+   - `/applicants/*`
+   - `/applications/*`
+   - `/applicationForms/*`
+   - `/canonicalFields/*`
```

Comment:
Hmm -- I'm surprised about how coupled these paths are to our API.  This one in particular refers to an endpoint that doesn't exist any more (as does `/applicants`), and we're missing some new ones (`/organizations` and `/baseFields`)

Do we need these redirect URIs at all?  What happens if we skip them entirely and only use the root URL?  Or just have `/*`?

It would be great to not need to have developers manually update keycloak config every time we add a new API endpoint.

Reply:
Agreed! I suppose I was thinking about these paths as generic examples, but that doesn't really make any sense